main.go

```diff
@@ -219,8 +242,10 @@ func watchTriggers() {
 		currQuote, err := types.ParseQuote(string(d.Body[:]))
 		failOnError(err, "Failed to parse Quote")
 		consoleLog.Debugf("New Quote for %s at price %s\n", currQuote.Stock, currQuote.Price.String())
-		buyTree, buyFound := autoTxStore[TreeKey{currQuote.Stock, "Buy"}] // tree, found
-		sellTree, sellFound := autoTxStore[TreeKey{currQuote.Stock, "Sell"}]
+		AutoTxStorage.Mutex.Lock()
+		buyTree, buyFound := AutoTxStorage.AutoTxTrees[TreeKey{currQuote.Stock, "Buy"}] // tree, found
+		sellTree, sellFound := AutoTxStorage.AutoTxTrees[TreeKey{currQuote.Stock, "Sell"}]
+		AutoTxStorage.Mutex.Unlock()
 		if !buyFound && !sellFound {
```

Comment:
I think you should separate this check (execution continues if `buyTree` is `nil` but `sellTree` isn't).

This probably works in its current state because the iterators are smart enough to handle `nil` trees gracefully.